internal/consensus/reactor.go

```diff
@@ -421,31 +415,58 @@ func (conR *Reactor) subscribeToBroadcastEvents() {
 	const subscriber = "consensus-reactor"
 	if err := conR.conS.evsw.AddListenerForEvent(subscriber, types.EventNewRoundStep,
 		func(data cmtevents.EventData) {
-			conR.broadcastNewRoundStepMessage(data.(*cstypes.RoundState))
-			conR.updateRoundStateNoCsLock()
+			rs := data.(cstypes.RoundState)
+
+			// update reactor's view of round state
```

Comment:
A suggestion here is to create a method to update the local copy of the round state.

The changes on these four methods are the same:

1. Lock the local copy
2. Update it (the source is different in the former two than in the latter two)
3. Unlock the local copy

I am not sure if changing the order here has an impact. Notice that in the original code we broadcast the message, then update the local copy. Here the order is inverted.